### PR TITLE
.NET CI run unit tests for windows

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
Things to consider before you submit the PR:

* [ ] ~Has `CHANGELOG.md` been updated?~
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] ~Did you add unit tests if this modifies the Windows code?~
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Related PRs or issues

[AB#77064](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/77064)

### CODE REVIEW

**Comment by MatkovIvan** [comment]
(Rebase to the latest `develop` with some squashes)

File: Tests/Microsoft.AppCenter.Test.Windows/AppCenterTest.cs [hunk 1]
```diff
@@ -644,7 +644,16 @@ public void ParsingFailIfWhiteSpaceBeforeFirstSecret()
...
+            try
+            {
+                parsedSecret = AppCenter.GetSecretAndTargetForPlatform(secrets, platformId);
+                Assert.Fail("It should be failed.");
+            } 
+            catch (Exception e)
+            {
+                Assert.AreEqual(e.GetType(), typeof(AppCenterException));
+            }
             Assert.AreNotEqual(appSecret, parsedSecret);
```
Comment (guperrot) [inline Tests/Microsoft.AppCenter.Test.Windows/AppCenterTest.cs]: We don't need this line then, and we don't need to keep parsedSecret as a variable.

File: Tests/Microsoft.AppCenter.Test.Windows/CodeCoverage.runsettings [hunk 2]
```diff
@@ -8,11 +8,11 @@
           <CodeCoverage>
             <ModulePaths>
               <Include>
-                <ModulePath>.*\Microsoft.AppCenter.NET.dll$</ModulePath>
-                <ModulePath>.*\Microsoft.AppCenter.Analytics.NET.dll$</ModulePath>                        
+                <ModulePath>.*Microsoft\.AppCenter\.NET\.dll$</ModulePath>
+                <ModulePath>.*Microsoft\.AppCenter\.Analytics\.NET\.dll$</ModulePath>
```
Comment (guperrot) [inline Tests/Microsoft.AppCenter.Test.Windows/CodeCoverage.runsettings]: Can we enable that for more projects?

File: Tests/Microsoft.AppCenter.Test.Windows/CodeCoverage.runsettings [hunk 3]
```diff
@@ -8,11 +8,11 @@
...
             <ModulePaths>
               <Include>
-                <ModulePath>.*\Microsoft.AppCenter.NET.dll$</ModulePath>
-                <ModulePath>.*\Microsoft.AppCenter.Analytics.NET.dll$</ModulePath>                        
+                <ModulePath>.*Microsoft\.AppCenter\.NET\.dll$</ModulePath>
+                <ModulePath>.*Microsoft\.AppCenter\.Analytics\.NET\.dll$</ModulePath>                       
               </Include>
               <Exclude>
-                <ModulePath>.*Test.Windows.dll</ModulePath>
+                <ModulePath>.*Test\.Windows\.dll$</ModulePath>
```
Comment (guperrot) [inline Tests/Microsoft.AppCenter.Test.Windows/CodeCoverage.runsettings]: Do we know why we need that?

File: Tests/Microsoft.AppCenter.Test.Windows/Storage/MockStorage.cs [hunk 4]
```diff
@@ -47,6 +47,7 @@ public Task DeleteLogs(string channelName)
         {
             lock (this)
             {
+                AppCenterLog.Debug(AppCenterLog.LogTag, $"Storage.DeleteLogs for channelName={channelName}");
```
Comment (guperrot) [inline Tests/Microsoft.AppCenter.Test.Windows/Storage/MockStorage.cs]: Is this actually printed in test logs? Just curious.

File: Tests/Microsoft.AppCenter.Test.Windows/TestLog.cs [hunk 5]
```diff
@@ -14,6 +14,9 @@ public class TestLog : LogWithProperties
     {
         internal const string JsonIdentifier = "testlog";
 
+        private static int DebugId;
+        private int _debugId = DebugId++;
```
Comment (guperrot) [inline Tests/Microsoft.AppCenter.Test.Windows/TestLog.cs]:
```suggestion

        private int _debugId = DebugId++;
```

File: Tests/Microsoft.AppCenter.Test.Windows/Storage/MockStorage.cs (hunk 4 above)
Comment (MatkovIvan) [reply Tests/Microsoft.AppCenter.Test.Windows/Storage/MockStorage.cs]: yes

**Pull request merged**